tools/src/dynamic-macros/macros.ts

```diff
@@ -165,15 +165,24 @@ export default {
     const url = await UrlUtils.constructManifestUrlAsync(path.join(EXPO_DIR, pathToHome));
 
     try {
-      const manifest = await getManifestAsync(url, platform, null);
+      const classicOrModernManifest = (await getManifestAsync(url, platform, null)) as any;
+      let classicManifest: Manifest;
+      if (classicOrModernManifest.launchAsset) {
+        // hack to convert modern manifest to classic
+        const bundleUrl = classicOrModernManifest.launchAsset.url;
+        classicManifest = classicOrModernManifest.extra.expoClient;
+        (classicManifest as any).bundleUrl = bundleUrl;
+      } else {
+        classicManifest = classicOrModernManifest;
+      }
 
-      if (manifest.name !== 'expo-home') {
+      if (classicManifest.name !== 'expo-home') {
```

Comment:
The dynamic macros are now used only by us to build Expo Go. I'm almost sure we can just assume it's the modern manifest and so do `manifest.extra?.expoClient?.name` here.